src/libraries/System.Private.CoreLib/src/System/IO/StreamReader.cs

```diff
@@ -865,65 +865,73 @@ private int ReadBuffer(Span<char> userBuffer, out bool readToUserBuffer)
             return task;
         }
 
-        private async Task<string?> ReadLineAsyncInternal()
+        private Task<string?> ReadLineAsyncInternal()
         {
-            if (_charPos == _charLen && (await ReadBufferAsync(CancellationToken.None).ConfigureAwait(false)) == 0)
+            async Task<string?> ReadInternalFast()
```

Comment:
this method is called `Fast` despite that it does more work than the other one. 

I think that we could name it by describing what it does: fetching data when the buffer is empty or all buffered data has been already consumed.

```suggestion
            async Task<string?> NoDataInTheBuffer()
```

Reply:
Done